server/tso/tso.go

```diff
@@ -56,8 +56,10 @@ type timestampOracle struct {
 	// TODO: remove saveInterval
 	saveInterval  time.Duration
 	maxResetTSGap func() time.Duration
-	tso           unsafe.Pointer
-	lastSavedTime atomic.Value
+	// tso info stored in the memory
+	mux           sync.RWMutex
+	tso           *tsoObject
```

Comment:
What about let tsoObject hold mutex like following:
```golang
type tsoObject struct {
    sync.RwMutex
    phsical time.Time
    logical int64
}
```

and Provide `get` and `set` function controlled by rwMutex? That would be much readable.

Reply:
Good idea.

Reply:
I think the 2 are not the same.

Reply:
I realize that we can not do this, otherwise, it will be meaningless to use `sync.RWMutex` instead of `atomic`.

Reply:
good catch